packages/instant/src/util/heartbeater_factory.ts

```diff
@@ -0,0 +1,22 @@
+import { asyncData } from '../redux/async_data';
+import { Store } from '../redux/store';
+
+import { Heartbeater } from './heartbeater';
+
+export interface HeartbeatFactoryOptions {
+    store: Store;
+    shouldPerformImmediatelyOnStart: boolean;
+}
+export const generateAccountHeartbeater = (options: HeartbeatFactoryOptions): Heartbeater => {
+    const { store, shouldPerformImmediatelyOnStart } = options;
+    return new Heartbeater(async () => {
```

Comment:
There's a small bug in my branch that i fixed that will have a conflict here. but we should not fetch account info if `state.providerState.account.state === AccountState.None` perhaps this check belongs in the provider component and we just don't start the heartbeater, or this function returns undefined in that case and isolate reach into the state in here